internal/csi-addons/rbd/replication.go

```diff
@@ -284,9 +284,9 @@ func (rs *ReplicationServer) EnableVolumeReplication(ctx context.Context,
 	if err != nil {
 		switch {
 		case errors.Is(err, corerbd.ErrImageNotFound):
-			err = status.Errorf(codes.NotFound, "volume %s not found", volumeID)
+			err = status.Errorf(codes.NotFound, err.Error())
```

Comment:
code duplication for `NotFound` error cases, consider using `fallthrough`:

```go
switch {
    case errors.Is(err, corerbd.ErrImageNotFound):
        fallthrough
    case errors.Is(err, util.ErrPoolNotFound):
        err = status.Errorf(codes.NotFound, err.Error())
...
```

this code block is repeated really a lot, it is probably worth it to put it in a helper function.